omnidoc/lark_tree_matcher.py

```diff
@@ -0,0 +1,154 @@
+"""Tree matcher based on Lark grammar
+
+TODO: Move to lark-parser, refactor reconstruct.py to use this as base class.
+"""
+
+
+from collections import defaultdict
+
+from lark.tree import Tree
+from lark.common import ParserConf
+from lark.lexer import Token
+from lark.parsers import earley
+from lark.grammar import Rule, Terminal, NonTerminal
+
+
+def is_discarded_terminal(t):
+    return t.is_term and t.filter_out
+
+def is_iter_empty(i):
+    try:
+        _ = next(i)
```

Comment:
needn't assingn to `_`. Just do `next(i)` I guess?

Reply:
Pylint doesn't like it when I ignore return values.

Is it against pep8?

Reply:
Not really. using `_` is not recommended [[1]](https://xkcd.com/285/). I think ignoring return value of next(i) is fine.